File: checkout-api/src/main/groovy/ish/oncourse/willow/service/CorporatePassApi.java
package ish.oncourse.willow.service;

import ish.oncourse.willow.model.checkout.CheckoutModelRequest;
import ish.oncourse.willow.model.checkout.corporatepass.CorporatePass;
import ish.oncourse.willow.model.checkout.corporatepass.GetCorporatePassRequest;
import ish.oncourse.willow.model.checkout.corporatepass.MakeCorporatePassRequest;
import ish.oncourse.willow.model.common.CommonError;
import ish.oncourse.willow.model.common.ValidationError;

import java.io.InputStream;
import java.io.OutputStream;
import java.util.List;
import java.util.Map;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.MediaType;

@Path("/")
public interface CorporatePassApi  {

    @POST
    @Path("/v1/getCorporatePass")
    @Produces({ "application/json" })
    @CollegeInfo
    CorporatePass getCorporatePass(GetCorporatePassRequest request);

    @GET
    @Path("/v1/isCorporatePassEnabled")
    @Produces({ "application/json" })
    @CollegeInfo
    Boolean isCorporatePassEnabled();

    @POST
    @Path("/v1/isCorporatePassEnabledFor")
    @Produces({ "application/json" })
    @CollegeInfo
    Boolean isCorporatePassEnabledFor(CheckoutModelRequest checkoutModelRequest);

    @POST
    @Path("/v1/makeCorporatePass")
    @Produces({ "application/json" })
    @XValidate
    @CollegeInfo
    void makeCorporatePass(MakeCorporatePassRequest request);
}

File: checkout/src/js/http/CorporatePassApi.ts
import {HttpService} from "../common/services/HttpService";
import {CheckoutModelRequest} from "../model/checkout/CheckoutModelRequest";
import {CorporatePass} from "../model/checkout/corporatepass/CorporatePass";
import {GetCorporatePassRequest} from "../model/checkout/corporatepass/GetCorporatePassRequest";
import {MakeCorporatePassRequest} from "../model/checkout/corporatepass/MakeCorporatePassRequest";
import {CommonError} from "../model/common/CommonError";
import {ValidationError} from "../model/common/ValidationError";

export class CorporatePassApi {
  constructor(private http: HttpService) {
  }

  getCorporatePass(request: GetCorporatePassRequest): Promise<CorporatePass> {
    return this.http.POST(`/getCorporatePass`, request);
  }
  isCorporatePassEnabled(): Promise<boolean> {
    return this.http.GET(`/isCorporatePassEnabled`);
  }
  isCorporatePassEnabledFor(checkoutModelRequest: CheckoutModelRequest): Promise<boolean> {
    return this.http.POST(`/isCorporatePassEnabledFor`, checkoutModelRequest);
  }
  makeCorporatePass(request: MakeCorporatePassRequest): Promise<any> {
    return this.http.POST(`/makeCorporatePass`, request);
  }
}
